src/exe/model.cc

```diff
@@ -40,6 +41,99 @@
 namespace colmap {
 namespace {
 
+// Aligns the reconstruction to the plane defined by running PCA on the 3D
+// points. The model centroid is at the origin of the new coordinate system
+// and the X axis is the first principal component with the Y axis being the
+// second principal component
+void AlignToPrincipalPlane(Reconstruction& recon, SimilarityTransform3& tform) {
+  // Perform SVD on the 3D points to estimate the ground plane basis
+  auto centroid = recon.ComputeCentroid();
+  Eigen::MatrixXd points(3, recon.NumPoints3D());
+  int pidx = 0;
+  for (const auto point : recon.Points3D()) {
+    points.col(pidx++) = point.second.XYZ() - centroid;
+  }
+  Eigen::Matrix3d basis =
+      points.jacobiSvd(Eigen::ComputeThinU | Eigen::ComputeThinV).matrixU();
+  Eigen::Matrix3d rot_mat;
+  rot_mat << basis.col(0), basis.col(1), basis.col(0).cross(basis.col(1));
+  rot_mat.transposeInPlace();
+
+  tform = SimilarityTransform3(1.0, RotationMatrixToQuaternion(rot_mat),
+                               -rot_mat * centroid);
+
+  // if camera plane ends up below ground then flip basis vectors and create new
+  // transform
+  class Image* test_img = new class Image(recon.Images().begin()->second);
```

Comment:
memory leak 

I also don't see a reason for not creating the object on the stack?

Reply:
I think I was having a weird compilation issue here. I changed it to `Image test_img = recon.Images().begin()->second` and now it works fine. Thanks for pointing this out.